src/App.tsx

```diff
@@ -1,61 +1,129 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const todosWithUser: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState(todosWithUser);
+  const [userSelect, setUserSelect] = useState('0');
+  const [todoTitle, setTodoTitle] = useState('');
+  const [errorUserSelect, setErrorUserSelect] = useState(false);
+  const [errorTodoTitle, setErrorTodoTitle] = useState(false);
+
+  const getNextId = (actualTodos: Todo[]): number => {
+    const maxId = actualTodos.reduce(
+      (max, todo) => (todo.id > max ? todo.id : max),
+      0,
+    );
+
+    return maxId + 1;
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    let hasError = false;
+
+    if (userSelect === '0') {
+      setErrorUserSelect(() => true);
+      hasError = true;
+    }
+
+    if (todoTitle === '') {
+      setErrorTodoTitle(true);
+      hasError = true;
+    }
+
+    if (hasError) {
+      return;
+    }
```

Comment:
Instead of creating variables to keep track of errors and return if true, You can return in places where you assign it to `hasError`